grass/kernel.c

```diff
@@ -142,6 +142,11 @@ static int proc_try_send(struct process* sender) {
 
             dst->syscall.msg.status = RECEIVED;
             dst->syscall.msg.sender = sender->pid;
+          /* Student's code goes here (page table translation)
+           * sender->syscall's address space belongs to sender->pid's address space,
+           * In this case it needs to be translated to the dst->pid's address space
+           * before the message can be copied.
+           */
```

Comment:
I believe that this is no longer necessary. Please bring this up for discussion in our next meeting.